rust/arrow/src/array/mod.rs

```diff
@@ -121,8 +121,8 @@ pub use self::array_primitive::PrimitiveArray;
 pub use self::array_string::LargeStringArray;
 pub use self::array_string::StringArray;
 pub use self::array_struct::StructArray;
+pub use self::array_union::UnionArray;
```

Comment:
👍 

Looks like we could also do the same thing (move `NullArray` from `null.rs` to `null_array.rs`), though as a different PR. I think this PR is a good improvement itself.